source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -419,9 +514,16 @@ ServerSslSocketFactory::createTransportSocket(Network::TransportSocketOptionsCon
     absl::ReaderMutexLock l(&ssl_ctx_mu_);
     ssl_ctx = ssl_ctx_;
   }
+  bool match = false;
+  if (transport_socket_options != nullptr) {
+    match =
+        tlsKeyLogMatch(transport_socket_options->getLocal(), transport_socket_options->getRemote());
```

Comment:
I think this logic should move into the `SslSocket::setTransportSocketCallbacks` constructor, so that it is shared between the downstream and upstream code, and then from the transport socket callbacks you can get `connection().connectionInfoProvider().localAddress()` so that you don't need to add anything to transport socket options.

Reply:
fixed